instadam/models/annotation.py

```diff
@@ -23,7 +23,7 @@ class Annotation(db.Model):
 
     # TODO: consistency problem - image's project and project_id should be the consistent
     # backref: original_image
-    image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
```

Comment:
## Potential Consistency Problem (Could Deal With This Later)
### Current Relationship Between Annotation, Label, Project and Image
annotations -> label: many to one
label -> project: many to one
===thus===> annotation -> label -> project: many to one

annotation -> image: many to one
image -> project: many to one
===thus===> annotation -> project: many to one

### We Have To Maintain The Following Consistency
the project of the label of annotation = the project of the original_image of annotation = the project of annotation

### Issue With Maintaining The Consistency
Since currently everything is nullable (except for project_id for image), we might encounter problem with maintaining the consistency. Consider the following scenario:
1. An annotation has a **Label A** that has a **null project** (since project_id is nullable in **label**), and an **original_image a** with **project 1** 
2. Later, the **Label A** is specified with **project 2**
3. Now the annotation has a **Label A** belonging to **project 2**, while its **original_image a** belongs to **project 1**

### Proposed Solution
 A way to deal with it is to set nullable=False for all the fields (project_id, image_id, label_id in annotation model, and project_id in label model), and check for consistency before adding to the database. 
### Other Things To Notice
 project and image should be at the same time nullable or non-nullable